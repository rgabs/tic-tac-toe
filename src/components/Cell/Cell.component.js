import './Cell.style.css';
import PropTypes from 'prop-types';
import React from 'react';
import noop from 'lodash/noop';


const Cell = ({value, cellclickHandler}) =>
  <div className={`cell ${value ? 'disabled-cell' : '' }`} onClick={cellclickHandler} >{value}</div>;


Cell.propTypes = {
  'cellclickHandler': PropTypes.func.isRequired,
  'value': PropTypes.oneOf([
    'X',
    'O',
    ''
  ]).isRequired
};
Cell.defaultProps = {
  'cellclickHandler': noop,
  'value': ''
};

export default Cell;
