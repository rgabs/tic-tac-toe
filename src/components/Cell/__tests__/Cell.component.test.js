import Cell from '../Cell.component';
import React from 'react';
import noop from 'lodash/noop';
import renderer from 'react-test-renderer';
test('Cell: snapshot test', () => {
  const cell = 'X';
  const snapshot = renderer.create(<Cell value={cell}
    cellclickHandler={noop}/>).toJSON();
  expect(snapshot).toMatchSnapshot();
});
