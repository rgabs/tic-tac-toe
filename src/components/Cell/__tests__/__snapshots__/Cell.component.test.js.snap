// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell: snapshot test 1`] = `
<div
  className="cell disabled-cell"
  onClick={[Function]}
>
  X
</div>
`;
